pkg/kvstore/etcd.go

```diff
@@ -982,42 +1003,32 @@ reList:
 		}
 
 		e.limiter.Wait(ctx)
-		res, err := e.client.Get(ctx, w.Prefix, client.WithPrefix(),
-			client.WithSerializable())
+		revision, err := e.paginatedList(ctx, scopedLog, w.Prefix, func(key *mvccpb.KeyValue) {
```

Comment:
I am wondering if it wouldn't make sense to keep the earlier interface and still return the full list here. 
Now with multiple list calls to Etcd, there will be a higher chance of compaction happening in-between list calls so I would prefer not to make it dependent on the speed of processing events in `w.Events`.

Reply:
Fair point. I adopted this approach because etcd requests were automatically rate-limited based on the events processing speed, which to me made sense to limit the load on etcd in case of slow consumers. And it also limited the number of events that needed to be cached at this stage before being propagated. But it's also true that if compaction occurs in between that would lead to a higher load or even the risk of not making progress at all.